### PR TITLE
Mail template

### PR DESCRIPTION
**Pull request by maxhansson**

### CODE REVIEW

File: app/controllers/mail_templates_controller.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  end
+
+  def set_default
+    @mail_template.from = current_organization.email
+    @mail_template.to = 'sale.contact_email'
+    @mail_template.attachment = 'sale.invoice_number.pdf'
+  end
+
+  def set_variable
+    @mail_template.subject = 'Faktura från ' + current_organization.name
```
Comment (qzio) [inline app/controllers/mail_templates_controller.rb]: We should not default to Swedish, but rather use the translation system, (`"#{t(:invoice_from)} #{current_organization.name}"`)

File: app/models/mail_template.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  TEMPLATE_TYPES = ['invoice', 'reminder']
+
+  attr_accessible :name, :template_type, :from, :to, :subject, :attachment, :text
+
+  belongs_to :organization
+
+  validates :name, presence: true, uniqueness: { scope: :organization_id}
+  validates :template_type, inclusion: { in: TEMPLATE_TYPES }
+  validates :from, presence: true
+  validates :to, presence: true
```
Comment (qzio) [inline app/models/mail_template.rb]: Do we need `from`/`to` columns?
What are they used for?

File: app/views/invoice_mailer/invoice_email.text.erb [hunk 3]
```diff
@@ -1,6 +1,10 @@
-Invoice sent to you by <%= @organization.name %>
-=====================================================
+<% if @text %>
+  <%= @text %>
```
Comment (qzio) [inline app/views/invoice_mailer/invoice_email.text.erb]: For this to be functional, I think we need to enable the user to use some variables.

At least who made the invoice.

File: app/models/mail_template.rb (hunk 2 above)
Comment (maxhansson) [inline app/models/mail_template.rb]: When a template is registered, it gives the user a better understanding when the mails from / to is explicitly shown in the form. 
Take the mails from / to from the database may be beneficial on certain occasions.

**Comment by maxhansson** [comment]
The info-buttons location in mail-template form need styling. Also in issue #186.

File: app/models/mail_template.rb (hunk 2 above)
Comment (qzio) [inline app/models/mail_template.rb]: but `from` will always be the `organization` email right?
and `to` will always be the the receiver of the invoice.

I haven't tried the GUI, but maybe we don't need to have database columns for them?

`to` and `from` could just be `methods` on the model maybe?

```ruby
def to
  'sale.contact_email'
end

def from
  organization.email
end
```

then we wouldn't have to do any validations on them.

File: db/migrate/20170615122507_remove_columns_in_mail_templates.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class RemoveColumnsInMailTemplates  < ActiveRecord::Migration
```
Comment (qzio) [inline db/migrate/20170615122507_remove_columns_in_mail_templates.rb]: This is only neded if we create the columns in the first place.

It's needed on you computer only, so I suggest, in the Branch, remove the columsn from the previous migration-file, and then remove this whole file. It makes the filesystem a bit cleaner.

File: app/mailers/invoice_mailer.rb [hunk 5]
```diff
@@ -34,8 +35,10 @@ def set_text(organization, sale, mail_template)
     if !mail_template.text.include? "#"
       return mail_template.text
     elsif sale.user.contacts.present?
-      contact = organization.contacts.find(sale.user.contacts)
-      return set_variables(mail_template.text, Contact::ALLOWED_COLUMNS, contact.attributes)
+      sale_fields = sale.attributes
+      contact_fields = organization.contacts.find(sale.user.contacts).attributes
+      fields = sale_fields.merge(contact_fields)
```
Comment (qzio) [inline app/mailers/invoice_mailer.rb]: Hm... wouldn't this override the variables from contact_field?

I propose to use something like `ALLOWED_COLUMNS = ['#recipient_name', '#sender_name', '#sender_telephone' ...]` to make it extra clear for the user that will type in the variable *names* in their mail template.

**Pull request merged**